### ISSUE TITLE
Error when installing Istio-ingressgateway &  egressgateway in Ambient Mode

### ISSUE THREAD

**Issue by Micky-Yang** [istio/istio#53027]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

When I use the 1.23.0 istioctl tool to install and deploy Istio Ambient Mode on a Kind k8s cluster on a local Mac computer, Istio-ingress fails to start. From the logs, I find that the pilot-agent execution format inside the container is incorrect. I'm not sure if this is a problem with the istioctl tool or an image packaging problem. Can someone help me check it out?
![image](https://github.com/user-attachments/assets/e43862f1-659d-46f3-99da-fb8bc9fccc1a)
![image](https://github.com/user-attachments/assets/5445f97f-84ae-4a4f-bc89-1cb4f7683895)

### Version

```Text
➜  ~ istioctl version
client version: 1.23.0
control plane version: 1.23.0
data plane version: 1.23.0 (1 proxies)

➜  ~ kubectl version
Client Version: v1.29.0
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.0
```

### Additional Information

➜  ~ istioctl bug-report

Target cluster context: kind-1-28-0

Running with the following config:

istio-namespace: istio-system
full-secrets: false
timeout (mins): 30
include: {  }
exclude: { Namespaces: kube-node-lease,kube-public,kube-system,local-path-storage }
end-time: 2024-09-06 17:39:39.651195 +0800 CST

Cluster endpoint: https://127.0.0.1:62635
CLI version:
version.BuildInfo{Version:"1.23.0", GitRevision:"d0ca037e58f90f58b38b33c5769fec25b6bfa9f3", GolangVersion:"go1.22.6", BuildStatus:"Clean", GitTag:"1.23.0"}

The following Istio control plane revisions/versions were found in the cluster:
Revision default:
&version.MeshInfo{
    {
        Component: "pilot",
        Revision:  "default",
        Info:      version.BuildInfo{Version:"1.23.0", GitRevision:"d0ca037e58f90f58b38b33c5769fec25b6bfa9f3", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.23.0"},
    },
}

The following proxy revisions/versions were found in the cluster:
Revision default: Versions {1.23.0}

Fetching logs for the following containers:

default/podinfo/podinfo-5749df899b-gln6j/podinfo
istio-system/istio-cni-node/istio-cni-node-2hxnr/install-cni
istio-system/istio-egressgateway/istio-egressgateway-7b9688dc7c-nzk9t/istio-proxy
istio-system/istio-ingressgateway/istio-ingressgateway-78797b6b44-pbzzp/istio-proxy
istio-system/istiod/istiod-6bfbcfcdc8-vqxpx/discovery
istio-system/kiali/kiali-5df8897db6-q9qhw/kiali
istio-system/prometheus/prometheus-bbcb484d4-krz5t/prometheus-server
istio-system/prometheus/prometheus-bbcb484d4-krz5t/prometheus-server-configmap-reload
istio-system/ztunnel/ztunnel-sfq8c/istio-proxy

Fetching Istio control plane information from cluster.

Fetching CNI logs from cluster.

Running Istio analyze on all namespaces and report as below:
Analysis Report:
Info [IST0102] (Namespace flux-system) The namespace is not enabled for Istio injection. Run 'kubectl label namespace flux-system istio-injection=enabled' to enable it, or 'kubectl label namespace flux-system istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace knative-serving) The namespace is not enabled for Istio injection. Run 'kubectl label namespace knative-serving istio-injection=enabled' to enable it, or 'kubectl label namespace knative-serving istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace kube-node-lease) The namespace is not enabled for Istio injection. Run 'kubectl label namespace kube-node-lease istio-injection=enabled' to enable it, or 'kubectl label namespace kube-node-lease istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace kube-public) The namespace is not enabled for Istio injection. Run 'kubectl label namespace kube-public istio-injection=enabled' to enable it, or 'kubectl label namespace kube-public istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace kube-system) The namespace is not enabled for Istio injection. Run 'kubectl label namespace kube-system istio-injection=enabled' to enable it, or 'kubectl label namespace kube-system istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace local-path-storage) The namespace is not enabled for Istio injection. Run 'kubectl label namespace local-path-storage istio-injection=enabled' to enable it, or 'kubectl label namespace local-path-storage istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace test) The namespace is not enabled for Istio injection. Run 'kubectl label namespace test istio-injection=enabled' to enable it, or 'kubectl label namespace test istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0118] (Service kube-system/kube-dns) Port name metrics (port: 9153, targetPort: 9153) doesn't follow the naming convention of Istio port.
Time used for creating the tar file is 85.097305ms.
Cleaning up temporary files in /var/folders/0b/5w2ptk7x0zb4cs3_ptq8gbsw0000gn/T/bug-report.
Done.

**Comment by Micky-Yang** [comment]
isitoctl installation command:
`➜  ~ istioctl install --set profile=ambient --set "components.ingressGateways[0].enabled=true" \
      --set "components.ingressGateways[0].name=istio-ingressgateway" \
      --set "components.egressGateways[0].enabled=true" \
      --set "components.egressGateways[0].name=istio-egressgateway" \
      --skip-confirmation`

**Comment by howardjohn** [comment]
exec format error implies you have a mismatch of architecture typically. Istio images are multi-arch though, so it shouldn't really happen, but maybe you have a cached version of the wrong architecture so its not pulling a new one?

**Comment by Micky-Yang** [comment]
Thank you for your answer. As you said, I went into the Kind Node environment to check the current Image image and found that the Image architecture used was indeed the expected amd64 Linux. This puzzled me because the image architecture was correct. However, when I deleted this image and reinstalled it, it would get the Image again and it was successfully run, but when I checked the Image on the Node again, I found that its architecture was also amd64 Linux. So, I am not sure why it reported an error when it was installed for the first time, even though its Image architecture was indeed the expected amd64 Linux.


<img width="1362" alt="image" src="https://github.com/user-attachments/assets/96eacd21-5cfa-491d-b7a2-e6a86c416971">
<img width="1154" alt="image" src="https://github.com/user-attachments/assets/95a9c3d9-3383-4fa2-87a6-e0a762134be3">
<img width="975" alt="image" src="https://github.com/user-attachments/assets/17105400-e6e3-4d3e-b4f3-3535cf9fc77f">
<img width="1717" alt="image" src="https://github.com/user-attachments/assets/e6ada30c-aa9c-4868-850e-3c0314cf1816">
